docs/source/_static/rest-api.yml

```diff
@@ -858,9 +862,305 @@ paths:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/Share"
+      security:
+        - oauth2:
+            - read:shares
+
+  /shares/{owner}/{server_name}:
+    get:
+      summary: List server shares
+      description: |
+        List shares granting access to a single server
+        (new in 5.0)
+      parameters:
+        - $ref: "#/components/parameters/sharedServerOwner"
+        - $ref: "#/components/parameters/sharedServerName"
+        - $ref: "#/components/parameters/paginationOffset"
+        - $ref: "#/components/parameters/paginationLimit"
+      responses:
+        200:
+          description: The list of shares granting access to the given server
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: "#/components/schemas/PaginatedList"
+                  - type: object
+                    properties:
+                      items:
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/Share"
+      security:
+        - oauth2:
+            - read:shares
+    post:
+      summary: Grant shared access
+      description: |
+        Grant shared access to a single server
+        (new in 5.0)
+      parameters:
+        - $ref: "#/components/parameters/sharedServerOwner"
+        - $ref: "#/components/parameters/sharedServerName"
+      requestBody:
+        description: The new group properties, as a JSON dict.
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                # expires_in not implemented, for now at least
+                # expires_in:
+                #   type: number
+                #   description: |
+                #     expiration in seconds.
+                #     If unspecified, no expiration.
+                group:
+                  type: string
+                  description: |
+                    group to grant permissions to.
+                    Exactly one of 'user' and 'group' can be specified.
+                user:
+                  type: string
+                  description: |
+                    user to grant permissions to.
+                    Exactly one of 'user' and 'group' can be specified.
+                scopes:
+                  type: array
+                  description: scopes to grant
+                  items:
+                    type: string
+        required: true
+      x-codegen-request-body-name: body
+      responses:
+        200:
+          description: The updated Share permissions
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Share"
+      security:
+        - oauth2:
+            - shares
+            - read:users:name
+            - read:groups:name
+    patch:
+      summary: Revoke shared access
```

Comment:
It's not modify, since it's explicitly and exclusively revocation. But on a per-user per-scope basis. But it does point to the fact that the delete method, which does exist and revokes all shared access to the server is missing here. I've added that and expanded the description of patch (summaries must stay very short to render nicely)